Gemfile

```diff
@@ -10,3 +10,5 @@ group :development do
   gem "method_profiler"
   gem "ruby-prof"
 end
+
+gem "memoizable"
```

Comment:
I'd also strongly advise against adding dependencies, unless they truly are needed. This doesn't seem like a case where it's worth it.